git_machete/utils.py

```diff
@@ -0,0 +1,244 @@
+
+    if (sys.platform == 'win32' or os.name == 'os2') and (ext != '.exe'):
+        executable = f"{executable}.exe"
+
+    if os.path.isfile(executable):
+        return executable
+
+    path = os.environ.get('PATH', os.defpath)
+    paths = path.split(os.pathsep)
+    for p in paths:
+        f = os.path.join(p, executable)
+        if os.path.isfile(f) and is_executable(f):
+            debug(cli_ctxt, f"find_executable({executable})", f"found {executable} at {f}")
+            return f
+    return None
+
+
+ENDC = '\033[0m'
+BOLD = '\033[1m'
+# `GIT_MACHETE_DIM_AS_GRAY` remains undocumented as for now,
+# was just needed for animated gifs to render correctly (`[2m`-style dimmed text was invisible)
+DIM = '\033[38;2;128;128;128m' if os.environ.get('GIT_MACHETE_DIM_AS_GRAY') == 'true' else '\033[2m'
+UNDERLINE = '\033[4m'
+GREEN = '\033[32m'
+YELLOW = '\033[33m'
+ORANGE = '\033[00;38;5;208m'
+RED = '\033[91m'
+
+ascii_only: bool = False
+
+
+def bold(s: str) -> str:
+    return s if ascii_only or not s else BOLD + s + ENDC
+
+
+def dim(s: str) -> str:
+    return s if ascii_only or not s else DIM + s + ENDC
+
+
+def underline(s: str, star_if_ascii_only: bool = False) -> str:
+    if s and not ascii_only:
+        return UNDERLINE + s + ENDC
+    elif s and star_if_ascii_only:
+        return s + " *"
+    else:
+        return s
+
+
+def colored(s: str, color: str) -> str:
+    return s if ascii_only or not s else color + s + ENDC
+
+
+fmt_transformations: List[Callable[[str], str]] = [
+    lambda x: re.sub('<b>(.*?)</b>', bold(r"\1"), x, flags=re.DOTALL),
+    lambda x: re.sub('<u>(.*?)</u>', underline(r"\1"), x, flags=re.DOTALL),
+    lambda x: re.sub('<dim>(.*?)</dim>', dim(r"\1"), x, flags=re.DOTALL),
+    lambda x: re.sub('<red>(.*?)</red>', colored(r"\1", RED), x, flags=re.DOTALL),
+    lambda x: re.sub('<yellow>(.*?)</yellow>', colored(r"\1", YELLOW), x, flags=re.DOTALL),
+    lambda x: re.sub('<green>(.*?)</green>', colored(r"\1", GREEN), x, flags=re.DOTALL),
+    lambda x: re.sub('`(.*?)`', r"`\1`" if ascii_only else UNDERLINE + r"\1" + ENDC, x),
+]
+
+
+def fmt(*parts: str) -> str:
+    result = ''.join(parts)
+    for f in fmt_transformations:
+        result = f(result)
+    return result
+
+
+def vertical_bar() -> str:
+    return "|" if ascii_only else u"│"
+
+
+def right_arrow() -> str:
+    return "->" if ascii_only else u"➔"
+
+
+def pretty_choices(*choices: str) -> str:
+    def format_choice(c: str) -> str:
+        if not c:
+            return ''
+        elif c.lower() == 'y':
+            return colored(c, GREEN)
+        elif c.lower() == 'yq':
+            return colored(c[0], GREEN) + colored(c[1], RED)
+        elif c.lower() in ('n', 'q'):
+            return colored(c, RED)
+        else:
+            return colored(c, ORANGE)
+    return f" ({', '.join(map_truthy_only(format_choice, choices))}) "
+
+
+def debug(cli_ctxt: CommandLineContext, hdr: str, msg: str) -> None:
```

Comment:
Yeah, since the only attributes of `CommandLineContext` accessed in `utils` are `opt_debug` and `opt_verbose`, I'd consider making them global variables for the sake of simplicity (hence `cli_ctxt` param will no longer be needed in many methods across other files as well)